transcode/transcode.go

```diff
@@ -174,10 +175,19 @@ func RunTranscodeProcess(transcodeRequest TranscodeSegmentRequest, streamName st
 	var mp4OutputsPre []video.OutputVideoFile
 	// Transmux received segments from T into a single mp4
 	if transcodeRequest.GenerateMP4 {
-		mp4TargetUrlBase, err := url.Parse(transcodeRequest.Mp4TargetUrl)
+		// Check if we should generate a standard MP4, fragmented MP4, or both.
+		mp4TargetUrlBase, enableStandardMp4, err := getMp4OutputType(transcodeRequest.Mp4TargetUrl)
 		if err != nil {
 			return outputs, segmentsCount, err
 		}
+		fragMp4TargetUrlBase, enableFragMp4, err := getMp4OutputType(transcodeRequest.FragMp4TargetUrl)
+		if err != nil {
+			return outputs, segmentsCount, err
+		}
+		if !(enableStandardMp4 || enableFragMp4) {
```

Comment:
In case both output URLs for mp4 and fmp4 is malformed somehow in the incoming request. There are additional layers of checks elsewhere in catalyst-api and probably Studio as well. I can remove if you think it's redundant but thought we keep an extra layer of security here.